keystorage/yubihsm2/src/main/java/tech/pegasys/signers/yubihsm/model/Opaque.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.yubihsm.model;
+
+import static java.nio.charset.StandardCharsets.US_ASCII;
+
+import tech.pegasys.signers.yubihsm.YubiHsmSession;
+
+import org.apache.tuweni.bytes.Bytes;
+
+/** @See <a href="https://developers.yubico.com/YubiHSM2/Commands/Get_Opaque.html">Opaque</a> */
+public class Opaque {
+  private final short opaqueId;
+
+  public enum OutputFormat {
+    ASCII,
+    HEX
+  }
+
+  public Opaque(final short opaqueId) {
```

Comment:
Seems unnecessary to have to create an instance of Opaque for every opaque data value in YubiHSM. Think the opaqueId should just be on the getOpaque method. Perhaps this class then becomes the top level API as well.

Reply:
yeah, let me re-think/re-tweak design slightly.